src/core/execute.c

```diff
@@ -453,6 +453,12 @@ int exec_spawn(
          * handoff timestamp. */
         dual_timestamp_now(&start_timestamp);
 
+        r = 0;
+        if (unit->manager->starting_ambient_set != 0 && MANAGER_IS_USER(unit->manager))
```

Comment:
can we conditionalize this further please, i.e. check if context->capability_ambient_set is non-zero? after all we won't need this in that case.

Reply:
I believe we need it. This patch is about transferring capabilities from systemd running as the session manager for a non-root user (`user@.service`) to sd-executor. `context->capability_ambient_set` is applied _by_ sd-executor and to do that it needs it's own capabilities in effective and permitted sets. sd-executor obtains them when during `execve()` the ambient set is properly prepared.

The alternative way would be to set file capabilities on sd-executor but that is both less flexible and less secure.

Indeed, we might set the ambient _for_ sd-executor to `context->capability_ambient_set`. This way systemd (`pid != 1`) needs to have all the capabilities required by services in the session, but sd-executor needs only those required by the service it will be starting. I will try that.